src/backend/src/services/reimbursement-requests.services.ts

```diff
@@ -128,6 +128,10 @@ export default class ReimbursementRequestService {
 
     if (!expenseType) throw new NotFoundException('Expense Type', expenseTypeId);
 
+    if (!expenseType.allowedRefundSources.includes(account)) {
+      throw new HttpException(400, 'Expense type is not paired with acceptable refund source account');
```

Comment:
Let's change this error message to: "The submitted refund source is not allowed to be used with the submitted expense type"